### DISCUSSION TITLE
Is ts -> rust compilation possible?

### DISCUSSION THREAD

**Discussion by Bessonov** [swc-project/swc#1628] [Ideas]
There are some projects which tired to use different targets already, for example https://github.com/ts-llvm/ts-llvm and linked related projects. I can imagine that beside of syntax there are challenges around natives like `fs`, `http` and so on. But, well, it could allow not only server-side execution, but also wasm and compilation of `tsc` to rust. What do you think?

**Comment by kdy1** [comment]
It would require really enormous amount of work.
If you send a PR, I would happily accept it and maintain it, but not by myself. I don't have time to do such enormous amount of work.

Also, I'm porting `tsc` with rust. But it will be a proprietary software, and I don't want to make enemy with js-to-rust compiler.
